planner/implementation/datasource.go

```diff
@@ -74,7 +74,7 @@ func (impl *TableReaderImpl) CalcCost(outCount float64, children ...memo.Impleme
 	reader := impl.plan.(*plannercore.PhysicalTableReader)
 	width := impl.tblColHists.GetAvgRowSize(impl.plan.SCtx(), reader.Schema().Columns, false, false)
 	sessVars := reader.SCtx().GetSessionVars()
-	networkCost := outCount * sessVars.NetworkFactor * width
+	networkCost := outCount * sessVars.GetNetworkFactor(nil) * width
```

Comment:
Why nil?

Reply:
TableReaderImpl don't have tableInfo property, so using nil to replace it. I will create another PR to fix it.